### PR TITLE
[issue-324] Add examples from spdx-spec, various bugfixes

### PR DESCRIPTION
**Pull request by nicoweidner**
Fixes https://github.com/spdx/tools-python/issues/324

### CODE REVIEW

File: tests/utils_test.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def load_and_clean_json(location):
     data = json.loads(content)
 
     if 'creationInfo' in data:
-        del(data['creationInfo'])
+        del (data['creationInfo'])
```
Comment (armintaenzertng) [inline tests/utils_test.py:180]: Is `del` special somehow to earn a space between the next bracket?

File: tests/utils_test.py [hunk 2]
```diff
@@ -350,20 +355,20 @@ def package_to_dict(cls, package):
...
             ('sourceInfo', package.source_info),
-            ('downloadLocation', package.download_location),
+            ('downloadLocation', str(package.download_location)),
             ('homepage', package.homepage),
             ('originator', cls.entity_to_dict(package.originator)),
             ('supplier', cls.entity_to_dict(package.supplier)),
             ('licenseConcluded', cls.license_to_dict(package.conc_lics)),
             ('licenseDeclared', cls.license_to_dict(package.license_declared)),
-            ('copyrightText', package.cr_text),
+            ('copyrightText', str(package.cr_text)),
```
Comment (armintaenzertng) [inline tests/utils_test.py:364]: Are these additional `str` used to handle `NONE` or `NO_ASSERTION`? If so, aren't there other fields that need the same treatment, like `homepage`?

**Review by armintaenzertng — COMMENTED** [review]
Looks good, just a few remarks :)

File: tests/utils_test.py (hunk 1 above)
Comment (nicoweidner) [reply tests/utils_test.py:180]: Yeah, looks like it's a special reserved keyword. I didn't even notice this, autoformat did it...

File: tests/utils_test.py (hunk 2 above)
Comment (nicoweidner) [reply tests/utils_test.py:364]: Maybe. These are the ones that I saw in some cases. I have to admit, I didn't fully understand the issue here. It showed some differences where `NOASSERTION` was inside quotation marks on one side, and without quotation marks on the other. I debugged and noticed that in the latter case, e.g. `package.cr_text` was a `NoAssert` instance. However, its `str` and `repr` methods do return a string, so I am not quite sure why casting it explicitly as string made a slight difference.

That said, all of this is test utility code that will probably be replaced, so I was fine with just fixing the issues that I observed in any halfway-reasonable way.

Comment (armintaenzertng) [reply tests/utils_test.py:364]: fair enough

**Review by armintaenzertng — APPROVED** [review]

**Review by armintaenzertng — APPROVED** [review]

**Pull request merged**